#33 karte ausspielen

MaxAlex42 opened the pull request:
Broker Integration Test muss noch hinzugefügt werden. Anmerkungen erwünscht :)

MaxAlex42 committed:
#33-Karte ausspielen

Grundgerüst für Karte ausspielen

MaxAlex42 committed:
#33-Karte ausspielen

Karte ausspielen Endpunkt und Logik hinzugefügt, Rundenkarten werden gespeichert, Broker Integration Test fehlt noch

MaxAlex42 committed:
#33-Karte ausspielen

Testcases für Code Coverage angepasst

MaxAlex42 committed:
#33-Karte ausspielen

"@Data" von Card entfernt

RNeumann12 commented on src/main/java/at/aau/serg/websocketdemoserver/gamelogic/LobbyManager.java:
```diff
@@ -97,4 +100,20 @@ public void dealNewRound(String code) throws Exception {
         lobby.getDeck().dealNewRound(lobby.getPlayers());
     }
+
+    public void cardPlayed(CardPlayRequest cardPlayRequest) {
```
Name ist etwas verwirrend. 
Sollte immer aussagen, was genau passiert bei der Methode.
playCardForPlayer (oder so ähnlich) wäre vielleicht passender

RNeumann12 commented on src/main/java/at/aau/serg/websocketdemoserver/gamelogic/LobbyManager.java:
```diff
@@ -97,4 +100,20 @@ public void dealNewRound(String code) throws Exception {
     }
+
+    public void cardPlayed(CardPlayRequest cardPlayRequest) {
+        Lobby l = lobbies.get(cardPlayRequest.getLobbyCode());
```
l ist kein guter Variablen Name (auch wenn es bei so kurzen Code relativ egal ist)

RNeumann12 commented on src/test/java/at/aau/serg/websocketdemoserver/WebSocketBrokerIntegrationTest.java:
```diff
@@ -193,6 +196,64 @@ public void testWebSocket_GetPlayerTurnUpdate() throws Exception {
+    /*
+    @Test
+    void testPlayCardEndpoint() throws Exception {
+        String userID = "123";
```
Setup, damit man überhaupt eine Karte spielen kann ist schon realtiv groß hier.
Vielleicht sollten wir eine initPlayableLobby (oder ähnlich) Methode machen. Damit wir hier wirklich nur auf das Karte spielen testen müssen

MaxAlex42 replied:
Hab jetzt eine Art Lobby Setup geschrieben, hab irgendwie ein selstames Issue wo ich einen String in "messages" habe der da nicht sein sollte, muss ich mir weiter anschauen

RNeumann12 commented on src/test/java/at/aau/serg/websocketdemoserver/WebSocketBrokerIntegrationTest.java:
```diff
@@ -193,6 +196,64 @@ public void testWebSocket_GetPlayerTurnUpdate() throws Exception {
+        StompSession sessionCardPlay = initStompSession(WEBSOCKET_TOPIC_CARD_PLAYED_RESPONSE);
+        sessionCardPlay.send(WEBSOCKET_TOPIC_PLAY_CARD, jsonObject);
+        String playCardResponse = messages.poll(1, TimeUnit.SECONDS);
+        assertThat(playCardResponse).isNotEmpty();
```
Hier könnte man noch irgendetwas abfragen. (z.B. eben ob der Response wirklich die Karte ist die man gespielt hat)

RNeumann12 reviewed (changes requested):
Sonst gut gelöst. 
Wenn die Integration Tests (und im Idealfall die Kleinigkeiten zur Bennenung) gefixt sind passt es.

MaxAlex42 committed:
#33-Karte ausspielen

Naming issues resolved wie besprochen, Integration Test, seltsamer Fehlerm möglicherweise Timing Problem?

RNeumann12 commented:
Passt. Alle weitere Anpassungen am besten in einem eigenen Issue / Branch
